openwisp_controller/config/tests/test_vpn.py

```diff
@@ -325,3 +327,44 @@ def test_auto_create_cert_with_long_device_name(self):
         cert = Cert.objects.filter(organization=org, name=device_name)
         self.assertEqual(cert.count(), 1)
         self.assertEqual(cert.first().common_name, client._get_common_name())
+
+    def test_placeholder_dh(self):
+        vpn = self._create_vpn(dh='')
+        self.assertIsNotNone(vpn.dh)
+        self.assertEqual(vpn.dh, Vpn._placeholder_dh)
+
+    @mock.patch.object(Vpn, 'dhparam')
+    def test_update_vpn_dh(self, dhparam):
+        dhparam.return_value = self._dh
+        vpn = self._create_vpn(dh='')
+        create_vpn_dh.delay(vpn.pk)
```

Comment:
I think it would be better to move this test to the transaction test case and avoid calling the celery task explicitly, it's more like a functional test